### ISSUE TITLE
Best way for filtering extremely large values ?

### ISSUE THREAD

**Issue by centminmod** [GoogleChrome/web-vitals#162]
I've setup Web Vitals JS to send metrics to Google Analytics and as a data source to Google Data studio. But today I received a TTFB event value for 40,000,000 ms ! As you can imagine that will skew all averages reported now. I can set custom filters in Google Data studio to exclude event value or avg value if >100000 or something like that. 

But is there a better way to filter at Web Vitals' JS library level at collection time ?

![image](https://user-images.githubusercontent.com/5899584/122501647-6ff4eb00-d038-11eb-94c0-13958a3bb1cd.png)

**Comment by philipwalton** [comment]
> But today I received a TTFB event value for 40,000,000 ms

Are you able to see what device/browser that value was reported for?

> is there a better way to filter at Web Vitals' JS library level at collection time ?

You could do something like this:

```js
getTTFB((metric) => {
  if (metric.value < 100000) {
    // Only process the result if the value is realistic...
  }
});
```

However, note that 100,000 may not be such an unrealistic value. 100000 ms is 100 seconds, which could potentially be explained by some sort of network issue where the connection just hung for a minute or so before resolving.

I do agree though that 40,000,000 is probably a bug in the implementation of the Navigation Timing API and not a real result.

If we get lots of reports of this happening, then I'd consider adding this check to the library, but I'm hesitant to do that yet since I haven't seen these types of results in the data for any of my own sites.

> As you can imagine that will skew all averages reported now.

Yeah, this is one of the reason why we specifically [recommend looking at the 75th percentile](https://web.dev/defining-core-web-vitals-thresholds/#choice-of-percentile) rather than using averages—it's less affected by outliers.

**Comment by DavidReider** [comment]
Hello, just wanted to add on that I also have recently received an unreasonably high TTFB score (4,294,970,000). I am unsure if it may be a similar issue as here or not - so far in our data it has only been reported this high one time, the rest are realistic times.

I do know that it came from a desktop user, but I am unsure which OS/Browser it may have come from. 

For right now, we have now implemented your suggestion of checking if the value is realistic and taking no action if it is above 5 minutes. 

If there is any other information I can provide, please feel free to let me know and I'll do my best to provide that. 

Thanks.

**Comment by philipwalton** [comment]
This should be fixed in [v2.1.2](https://github.com/GoogleChrome/web-vitals/blob/main/CHANGELOG.md#v212-2021-10-11).
